scripts/sct_maths.py

```diff
@@ -526,233 +467,52 @@ def get_data_or_scalar(argument, data_in):
     return data_out
 
 
-def concatenate_along_4th_dimension(data1, data2):
-    """
-    Concatenate two data along 4th dimension.
-    :param data1: 3d or 4d array
-    :param data2: 3d or 4d array
-    :return data_concat: concate(data1, data2)
-    """
-    if len(np.shape(data1)) == 3:
-        data1 = data1[..., np.newaxis]
-    if len(np.shape(data2)) == 3:
-        data2 = data2[..., np.newaxis]
-    return np.concatenate((data1, data2), axis=3)
-
-
-def denoise_nlmeans(data_in, patch_radius=1, block_radius=5):
+def convert_list_str(string_list, type='int'):
     """
-    data_in: nd_array to denoise
-    for more info about patch_radius and block radius, please refer to the dipy website: http://nipy.org/dipy/reference/dipy.denoise.html#dipy.denoise.nlmeans.nlmeans
+    Receive a string and then converts it into a list of selected type.
+    Example: "2,2,3" --> [2, 2, 3]
+    :param string_list: List of comma-separated string
+    :param type: string: int, float
+    :return:
     """
-    from dipy.denoise.nlmeans import nlmeans
-    from dipy.denoise.noise_estimate import estimate_sigma
-    from numpy import asarray
-    data_in = asarray(data_in)
-
-    block_radius_max = min(data_in.shape) - 1
-    block_radius = block_radius_max if block_radius > block_radius_max else block_radius
-
-    sigma = estimate_sigma(data_in)
-    denoised = nlmeans(data_in, sigma, patch_radius=patch_radius, block_radius=block_radius)
+    new_type_list = (string_list).split(",")
+    for inew_type_list, ele in enumerate(new_type_list):
+        if type is "int":
+            new_type_list[inew_type_list] = int(ele)
+        elif type is "float":
+            new_type_list[inew_type_list] = float(ele)
 
-    return denoised
+    return new_type_list
 
 
-def smooth(data, sigmas):
+def compute_similarity(img1: Image, img2: Image, fname_out: str, metric: str, metric_full: str, verbose):
```

Comment:
that's cool, i didn't know about the type checking in input args. I'm wondering if this could be leveraged for sphinx doc (ie instead of specifying the type inside the docstrings)

Reply:
The're typehints, unfortunatlely they don't provide type checking, but they are useful to have. They can definitely be used by sphinx (see https://pypi.org/project/sphinx-autodoc-typehints/), I wanted to add them everywhere when updating the doc, but it seemed like a massive effort.